unison-core/src/Unison/DataDeclaration.hs

```diff
@@ -391,216 +395,17 @@ hashDecls0 decls =
 hashDecls
   :: (Eq v, Var v)
   => Map v (DataDeclaration' v a)
-  -> Names.ResolutionResult v a [(v, Reference, DataDeclaration' v a)]
+  -> Names.ResolutionResult v a [(v, Reference.Id, DataDeclaration' v a)]
 hashDecls decls = do
   -- todo: make sure all other external references are resolved before calling this
   let varToRef = hashDecls0 (void <$> decls)
+      varToRef' = second Reference.DerivedId <$> varToRef
       decls'   = bindTypes <$> decls
-      bindTypes dd = dd { constructors' = over _3 (Type.bindExternal varToRef) <$> constructors' dd }
+      bindTypes dd = dd { constructors' = over _3 (Type.bindExternal varToRef') <$> constructors' dd }
       typeNames0 = Names.names0 mempty
-                 $ Rel.fromList (first Name.fromVar <$> varToRef)
+                 $ Rel.fromList (first Name.fromVar <$> varToRef')
       -- normalize the order of the constructors based on a hash of their types
       sortCtors dd = dd { constructors' = sortOn hash3 $ constructors' dd }
       hash3 (_, _, typ) = ABT.hash typ :: Hash
   decls' <- fmap sortCtors <$> traverse (bindNames mempty typeNames0) decls'
   pure  [ (v, r, dd) | (v, r) <- varToRef, Just dd <- [Map.lookup v decls'] ]
-
```

Comment:
Moved to `Unison.Builtin.Decls` module

Reply:
(This comment refers to the hundreds of lines below this one; I didn't move just a single blank line to a different module.)